internal-dns/tests/basic_test.rs

```diff
@@ -142,6 +198,45 @@ pub async fn nxdomain() -> Result<(), anyhow::Error> {
             }
         },
     };
+}
+
+#[tokio::test]
+pub async fn empty_record() -> Result<(), anyhow::Error> {
+    let test_ctx = init_client_server("oxide.internal".into()).await?;
+    let client = &test_ctx.client;
+    let resolver = &test_ctx.resolver;
+
+    // records should initially be empty
+    let records = client.dns_records_get().await?;
+    assert!(records.is_empty());
+
+    // Add an empty DNS record
+    let name = DnsRecordKey { name: "devron.oxide.internal".into() };
+    client
+        .dns_records_set(&vec![DnsKv { key: name.clone(), records: vec![] }])
```

Comment:
Adding this call was the only way to test this behavior, but it raises a question for me - if we try setting a record with "records: vec![]", would we prefer that be an error at *set* time, rather than *get* time?

Reply:
I think disallowing setting empty records is probably a reasonable position to take. I still do like the idea of NXDOMAIN on empty record if that situation does arise for some reason. Perhaps a change later on down the line lets these empty records sneak in, we still want the NXDOMAIN response.

Reply:
Ack, I'll leave this as-is for now then. Since the API is internal, and we handle this behavior, it seems mostly harmless.